### PR TITLE
feature (#3211) Reply on comment

### PR DESCRIPTION
**Pull request by OlegVat**
Oleh Vatuliak
issue https://github.com/ita-social-projects/GreenCity/issues/3211

Summary of issue
Added request to add a reply to user's event comment

Summary of change
The user now can reply to their event comments

### CODE REVIEW

**Review by danylohlynsky — APPROVED** [review]

File: core/src/main/java/greencity/controller/EventCommentController.java [hunk 1]
```diff
@@ -148,4 +148,30 @@ public ResponseEntity<Object> delete(@PathVariable Long eventCommentId, @ApiIgno
...
+    @ApiOperation(value = "Add reply.")
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = HttpStatuses.CREATED, response = AddEventCommentDtoRequest.class),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND),
+    })
+    @PostMapping("{eventId}/saveReply/{parentCommentId}")
+    public void saveReply(@PathVariable Long eventId, @RequestParam String replyText,
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/EventCommentController.java]: Please add tests for `201` response code

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 2]
```diff
@@ -184,4 +184,26 @@ public void delete(Long eventCommentId, UserVO user) {
...
+     * @param eventId         id of {@link greencity.dto.event.EventVO} to add
+     *                        comment to.
+     * @param replyText       text of
+     *                        {@link greencity.dto.eventcomment.EventCommentVO}
+     *                        reply.
+     * @param userVO          - {@link UserVO} user who replied.
+     * @param parentCommentId {@link greencity.dto.event.EventVO} comment on which
+     *                        replied.
+     */
+    public void saveReply(Long eventId, String replyText,
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java]: `eventId` never used

File: service/src/test/java/greencity/service/EventCommentServiceImplTest.java [hunk 3]
```diff
@@ -228,4 +228,36 @@ void deleteCommentThatDoesntExistsThrowException() {
...
+        UserVO userVO = getUserVO();
+        Long parentCommentId = 1L;
+        Long eventId = 1L;
+
+        when(eventCommentRepo.findById(parentCommentId))
+            .thenReturn(Optional.ofNullable(ModelUtils.getEventComment()));
+
+        eventCommentService.saveReply(eventId, "text", userVO, parentCommentId);
+
+        verify(eventCommentRepo).save(any(EventComment.class));
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/EventCommentServiceImplTest.java]: Duplication. Lines 248 - 257 are the same as lines 234 - 243

Comment (OlegVat) [reply service/src/test/java/greencity/service/EventCommentServiceImplTest.java]: Done

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java (hunk 2 above)
Comment (OlegVat) [reply service/src/main/java/greencity/service/EventCommentServiceImpl.java]: Done

File: core/src/main/java/greencity/controller/EventCommentController.java (hunk 1 above)
Comment (OlegVat) [reply core/src/main/java/greencity/controller/EventCommentController.java]: Fixed documentation

**Review by innayashna — APPROVED** [review]

**Review by NataliaHlazova — APPROVED** [review]

**Pull request merged**